src/frequenz/client/dispatch/test/service.py

```diff
@@ -0,0 +1,259 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Mock classes for the dispatch api.
+
+Useful for testing.
+"""
+
+from dataclasses import dataclass
+from datetime import datetime, timezone
+
+import grpc
+import grpc.aio
+
+# pylint: disable=no-name-in-module
+from frequenz.api.dispatch.v1.dispatch_pb2 import Dispatch as PBDispatch
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    DispatchCreateRequest as PBDispatchCreateRequest,
+)
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    DispatchDeleteRequest,
+    DispatchGetRequest,
+    DispatchList,
+)
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    DispatchListRequest as PBDispatchListRequest,
+)
+from frequenz.api.dispatch.v1.dispatch_pb2 import DispatchUpdateRequest
+from google.protobuf.empty_pb2 import Empty
+
+from frequenz.client.base.conversion_helper import to_datetime as _to_dt
+
+# pylint: enable=no-name-in-module
+from frequenz.client.dispatch import Client
+from frequenz.client.dispatch._internal_types import DispatchCreateRequest
+from frequenz.client.dispatch.types import Dispatch
+
+__all__ = ["Service", "MockClient"]
+
+
+@dataclass
+class Service:
+    """Dispatch mock service for testing."""
+
+    dispatches: list[Dispatch]
+
+    last_update_request: DispatchUpdateRequest | None = None
+
+    _last_id: int = 0
+
+    def __init__(self) -> None:
+        """Initialize the mock service."""
+        self.dispatches = []
```

Comment:
Are defaults properly initialized even if you write your own `__init__`? Any reason not to do this? Which allow initialization passing a list of dispatches already?

```suggestion
    dispatches: list[Dispatch] = dataclasses.field(default_factory=list)

    last_update_request: DispatchUpdateRequest | None = None

    _last_id: int = 0
```